.github/workflows/build.yml

```diff
@@ -8,10 +8,10 @@ on:
         required: true
         type: boolean
   push:
-    paths: ['.github/workflows/**', 'CMakeLists.txt', 'Makefile', '**.h', '*.c', '**.cpp']
+    paths: ['.github/workflows/**', '**/CMakeLists.txt', '**/Makefile', '**/*.h', '**/*.c', '**/*.cpp']
   pull_request:
     types: [opened, synchronize, edited, reopened, review_requested, ready_for_review]
-    paths: ['CMakeLists.txt', 'Makefile', '**.h', '*.c', '**.cpp']
+    paths: ['**/CMakeLists.txt', '**/Makefile', '**/*.h', '**/*.c', '**/*.cpp']
```

Comment:
someone explain to me why they look the way they do.

Reply:
It's the [glob format](https://en.wikipedia.org/wiki/Glob_(programming))

Reply:
In plain english it says:
in all directories and subdirectories
CMakeLists.txt
Makefile
*.h
*.c
*.cpp

Reply:
and you cant just use `**` instead of `**/*` ? that feels kind of wrong

Reply:
> and you cant just use `**` instead of `**/*` ? that feels kind of wrong

Yeah, for example `**/Makefile` matches files named exactly `Makefile` , while `**Makefile` would also match `abcdefgMakefile` and so on. The `**/*` format is the proper format for recursing directories.